### PR TITLE
Mention require's case-sensitivity in documentation

### LINKED ISSUE #13579: Documentation for `require()` should be updated
currently, the [doc](http://docs.julialang.org/en/latest/stdlib/base/#Base.require) says:
>When searching for files, require first looks in the current working directory, then looks for package code under Pkg.dir(), then tries paths in the global array LOAD_PATH.

Since [this commit](https://github.com/JuliaLang/julia/commit/75ab3602fcaf224dfcc550dad684125936e4bc44) was merged, `require()` hasn't been looking in the current working directory at all.

### PR DESCRIPTION
**Pull request by sivapvarma**
fixes  [#13579](https://github.com/JuliaLang/julia/issues/13579)

### CODE REVIEW

File: base/docs/helpdb/Base.jl [hunk 1]
```diff
@@ -1489,7 +1489,7 @@ This function is typically used to load library code, and is implicitly called b
 load packages.
 
 When searching for files, `require` first looks for package code under `Pkg.dir()`, then tries
-paths in the global array `LOAD_PATH`.
+paths in the global array `LOAD_PATH`. `require` is case-sensitive  on all platforms including those with case-insensitive filesystems like macOS and Windows.
```
Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: "Mac OS"

Comment (Keno) [inline base/docs/helpdb/Base.jl]: Apple renamed it to macOS officially.

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: Also, I don't think Windows is case-insensitive, is it?

Comment (yuyichao) [inline base/docs/helpdb/Base.jl]: https://github.com/JuliaLang/julia/pull/13542

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: > Apple renamed it to macOS officially.

OK, my bad.

Comment (sivapvarma) [inline base/docs/helpdb/Base.jl]: NTFS **is** case-sensitive! I did not know that. 

From [Wikipedia](https://en.wikipedia.org/wiki/Case_sensitivity) 

> Current Windows file systems, like NTFS, are case-sensitive; that is a readme.txt and a Readme.txt can exist in the same directory. However, the API for file access in Windows applications is case-insensitive, which makes filenames case-insensitive from the application's point of view.

I am not sure whether I should keep windows.

Comment (nalimilan) [inline base/docs/helpdb/Base.jl]: IIUC, NTFS can be case-sensitive, but Windows doesn't use that feature, so it's good idea to keep it.

Comment (tkelman) [inline base/docs/helpdb/Base.jl]: You can change a setting to make your filesystem case sensitive, and Cygwin can use things that way, but Win32 generally doesn't differentiate.

Please wrap docstring lines.

Comment (sivapvarma) [inline base/docs/helpdb/Base.jl]: wrapped lines now.

File: base/docs/helpdb/Base.jl [hunk 2]
```diff
@@ -1489,7 +1489,9 @@ This function is typically used to load library code, and is implicitly called b
 load packages.
 
 When searching for files, `require` first looks for package code under `Pkg.dir()`, then tries
-paths in the global array `LOAD_PATH`.
+paths in the global array `LOAD_PATH`. `require` is case-sensitive  on all
+platforms including those with case-insensitive filesystems like macOS and
```
Comment (stevengj) [inline base/docs/helpdb/Base.jl]: You need a comma before the word "including", and "MacOS" is capitalized.

Comment (yuyichao) [inline base/docs/helpdb/Base.jl]: > "MacOS" is capitalized.

Not anymore....

**Pull request merged**